### PR TITLE
Fixed 'retain cycle' warnings in GRPCWrappedCall.m

### LINKED ISSUE #2210: Fix "retain cycle" warnings in GRPCWrappedCall.m
In GRPCWrappedCall.m, three callbacks have warnings that say "Capturing 'self' strongly in this block is likely to lead to a retain cycle."

### PR DESCRIPTION
**Pull request by murgatroid99**
This fixes #2210.

### CODE REVIEW

File: src/objective-c/GRPCClient/private/GRPCWrappedCall.m [hunk 1]
```diff
@@ -132,8 +132,12 @@ - (instancetype) initWithHandler:(void (^)(NSDictionary *))handler {
     grpc_metadata_array_init(&_headers);
     _op.data.recv_initial_metadata = &_headers;
     if (handler) {
+      // Prevent reference cycle with handler
```
Comment (jcanizales) [inline src/objective-c/GRPCClient/private/GRPCWrappedCall.m]: nit: with _handler

File: src/objective-c/GRPCClient/private/GRPCWrappedCall.m [hunk 2]
```diff
@@ -132,8 +132,12 @@ - (instancetype) initWithHandler:(void (^)(NSDictionary *))handler {
...
+      __weak typeof(self) weakSelf = self;
       _handler = ^{
-        NSDictionary *metadata = [NSDictionary grpc_dictionaryFromMetadataArray:_headers];
+        __strong typeof(self) strongSelf = weakSelf;
```
Comment (jcanizales) [inline src/objective-c/GRPCClient/private/GRPCWrappedCall.m]: No need to retain it, as `_handler` is only ever called within `-finish`, and `self`'s reference count is always > 0 when an instance method is being executed (the compiler makes sure of that).

So `weakSelf->_headers` in the next line is safe.

Same for the other two blocks.

Comment (murgatroid99) [inline src/objective-c/GRPCClient/private/GRPCWrappedCall.m]: XCode says it's an error to dereference a weak pointer.

Comment (jcanizales) [inline src/objective-c/GRPCClient/private/GRPCWrappedCall.m]: :( It should let us; we know it's safe. Fair enough...

File: src/objective-c/GRPCClient/private/GRPCWrappedCall.m (hunk 1 above)
Comment (murgatroid99) [inline src/objective-c/GRPCClient/private/GRPCWrappedCall.m]: Fixed

**Pull request merged**